src/components/TrackedReactFilePlayer/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useRef, useEffect, RefObject } from 'react'
+import FilePlayer, { FilePlayerProps } from 'react-player/file'
+
+import {
+  trackMediaReady,
+  trackMediaPlay,
+  trackMediaPause,
+  trackMediaBufferStart,
+  trackMediaBufferEnd,
+  trackMediaEnd,
+  trackMediaError,
+  trackMediaSeekEnd,
+  trackMediaPlaybackRateChange,
+  trackMediaPictureInPictureChange,
+  startMediaTracking,
+} from '@snowplow/browser-plugin-media'
+
+interface PlayerIdProps {
+  id: string
+  label?: string
+}
+
+// Taken from:
+// https://github.com/cookpete/react-player/blob/795b19614977fbe2b89f6fd14503d1bfb121a722/types/base.d.ts#L53
+interface PlayerError {
+  error: any
+  data?: any
+  hlsInstance?: any
+  hlsGlobal?: any
+}
+
+function playerData(
+  player?: RefObject<FilePlayer>,
+  label?: string
+): Record<string, any> | undefined {
+  if (player?.current) {
+    const internal = player.current.getInternalPlayer()
+    return {
+      label,
+      mediaType: 'video',
+      playerType: 'react-player/file',
+      currentTime: internal?.currentTime ?? 0,
+      duration: internal?.duration,
+      muted: internal?.muted,
+      loop: internal?.loop,
+      paused: internal?.paused,
+      ended: internal?.ended,
+      playbackRate: internal?.playbackRate,
+      volume: internal?.muted ?? false ? 0 : internal?.volume * 100,
+    }
+  }
+}
+
+/**
+ * A wrapped version of the FilePlayer component that tracks media events
+ * using the Snowplow Media Tracking plugin.
+ * @returns FilePlayer
+ */
+export default function TrackedFilePlayer(
+  props: FilePlayerProps & PlayerIdProps
+): JSX.Element {
+  const { id, label } = props
+
+  // Ensure that an `id` prop is provided to prevent bad events
+  if (!id) {
+    throw new Error('TrackedFilePlayer requires an `id` prop')
+  }
+
+  const playerRef = useRef<FilePlayer>(null)
+
+  useEffect(() => {
+    startMediaTracking({ id })
```

Comment:
The `id` of the media tracking needs to be unique for each viewing. If I understand correctly, here it is the same for each player. Ideally this could generate a new UUID each time the player is initialized.

Reply:
Makes sense, have updated to use a UUID instead.